selfdrive/ui/qt/widgets/drive_stats.cc

```diff
@@ -43,19 +43,23 @@ QLayout* build_stat(QString name, int stat) {
   return layout;
 }
 
-void DriveStats::parseError(QString response) {
-  clearLayouts(vlayout);
-  vlayout->addWidget(new QLabel("No Internet connection"), 0, Qt::AlignCenter);
-}
+void DriveStats::parseResponse(QString response, bool save) {
+  static QString prev_response;
+  response = response.trimmed();
+  if (prev_response == response) {
```

Comment:
hmm, we shouldn't need to do this. maybe build the layout in the constructor and just update the text values in this function?

Reply:
updating the text should be super cheap, Qt probably already does this if when updating values